### PR TITLE
Hide delegates, model parameters constructors and other small things

### LINKED ISSUE #1974: Internalize signature delegates in Core and Data assemblies
We have these "signature" delegates throughout the codebase, used for purposes of entry-points and other dependency injection style things.

Many are public, as we see here.

https://github.com/dotnet/machinelearning/blob/7e21afa4612ddca53bbf9d67b84df2b58d4636c2/src/Microsoft.ML.Data/Dirty/ILoss.cs#L49

Yet there is no need for them to be public, and many of these have already been "internalized," e.g.:

https://github.com/dotnet/machinelearning/blob/7e21afa4612ddca53bbf9d67b84df2b58d4636c2/src/Microsoft.ML.Core/Prediction/ITrainer.cs#L20

We should complete this work, since we do not want these as part of the public surface of the API.

### PR DESCRIPTION
**Pull request by Ivanidzo4ka**
should fix #1974

### CODE REVIEW

**Comment by artidoro** [comment]
>     public sealed class TransformWrapper : ITransformer

I can't wait for this to go away!

---
Refers to: src/Microsoft.ML.Data/DataLoadSave/TransformWrapper.cs:20 in 622543c. [](commit_id = 622543c2c513eb412349573026a228c65409aa1f, deletion_comment = False)

**Comment by artidoro** [comment]
>             public readonly bool TextKeyValues;

You used to add doc comments in your previous PRs :`(

---
Refers to: src/Microsoft.ML.Data/Transforms/ValueToKeyMappingEstimator.cs:42 in 622543c. [](commit_id = 622543c2c513eb412349573026a228c65409aa1f, deletion_comment = False)

**Comment by Ivanidzo4ka** [comment]
>     public sealed class TransformWrapper : ITransformer

All we need to do is untangle that mess in word bag extraction.

---
In reply to: [466601135](https://github.com/dotnet/machinelearning/pull/2701#issuecomment-466601135) [](ancestors = 466601135)

---
Refers to: src/Microsoft.ML.Data/DataLoadSave/TransformWrapper.cs:20 in 622543c. [](commit_id = 622543c2c513eb412349573026a228c65409aa1f, deletion_comment = False)

**Comment by Ivanidzo4ka** [comment]
>             public readonly bool TextKeyValues;

Purpose of this PR is to hide stuff.
More we hide, less to document :D

I do want to change all this `Term`, `MaxNumKeys` property to something more appropriate, but not in this PR.

---
In reply to: [466601428](https://github.com/dotnet/machinelearning/pull/2701#issuecomment-466601428) [](ancestors = 466601428)

---
Refers to: src/Microsoft.ML.Data/Transforms/ValueToKeyMappingEstimator.cs:42 in 622543c. [](commit_id = 622543c2c513eb412349573026a228c65409aa1f, deletion_comment = False)

File: src/Microsoft.ML.Data/DataLoadSave/TransformWrapper.cs [hunk 1]
```diff
@@ -168,26 +168,4 @@ public SchemaShape GetOutputSchema(SchemaShape inputSchema)
             return SchemaShape.Create(transformer.GetOutputSchema(fakeSchema));
         }
     }
-
-    /// <summary>
-    /// Estimator for untrained wrapped transformers.
-    /// </summary>
-    public abstract class TrivialWrapperEstimator : TrivialEstimator<TransformWrapper>
```
Comment (abgoswam) [inline src/Microsoft.ML.Data/DataLoadSave/TransformWrapper.cs]: >TrivialWrapperEstimator [](start = 26, length = 23)

where did this go ?

Comment (abgoswam) [reply src/Microsoft.ML.Data/DataLoadSave/TransformWrapper.cs]: ah i see. no usages in code :)

---
In reply to: [259595477](https://github.com/dotnet/machinelearning/pull/2701#discussion_r259595477) [](ancestors = 259595477)

Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Data/DataLoadSave/TransformWrapper.cs]: To valhalla!

**Pull request merged**